front/app/components/UI/FileRepositorySelectAndUpload/components/FileSelectOrUploadModal.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import { Box, Button, Select, Text } from '@citizenlab/cl2-component-library';
+import { useParams } from 'react-router-dom';
+import { UploadFile } from 'typings';
+
+import { IFileAttachmentData } from 'api/file_attachments/types';
+import { IFileData } from 'api/files/types';
+import useFiles from 'api/files/useFiles';
+
+import useFeatureFlag from 'hooks/useFeatureFlag';
+
+import Modal from 'components/UI/Modal';
+
+import { useIntl } from 'utils/cl-intl';
+
+import FileInput from '../../FileUploader/FileInput';
+
+import messages from './messages';
+
+type Props = {
+  id: string;
+  onFileAdd: (fileToAdd: UploadFile) => void;
+  onFileAttach?: (fileToAttach: IFileData) => void;
+  fileAttachments?: IFileAttachmentData[];
+  maxSizeMb?: number;
+  dataCy?: string;
+};
+
+const FileSelectOrUploadModal = ({
+  id,
+  onFileAdd,
+  onFileAttach,
+  maxSizeMb,
+  fileAttachments,
+  dataCy,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const [modalOpen, setModalOpen] = React.useState(false);
+  const isDataRepositoryEnabled = useFeatureFlag({
+    name: 'data_repository',
+  });
+
+  const { projectId } = useParams();
+
+  const { data: files } = useFiles({
+    project: [projectId || ''],
+    enabled: !!projectId,
+  });
+
+  // Generate options for the file select dropdown
+  const fileOptions = files
+    ? files.data.map((file) => ({
+        value: file.id,
+        label: file.attributes.name,
+      }))
+    : [];
+
+  // Filter out already attached files
+  const filteredFileOptions = fileAttachments
+    ? fileOptions.filter(
+        (option) =>
+          !fileAttachments.find(
+            (attachment) =>
+              attachment.relationships.file.data.id === option.value
+          )
+      )
+    : fileOptions;
+
+  const handleFileOnAdd = (fileToAdd: UploadFile) => {
+    onFileAdd(fileToAdd);
+    setModalOpen(false);
+  };
+
+  return (
+    <>
+      <Button
+        onClick={() => setModalOpen(true)}
+        buttonStyle="secondary"
+        icon="file-add"
+      >
+        {formatMessage(messages.addFile)}
+      </Button>
+      <Modal
+        closeOnClickOutside={true}
+        close={() => setModalOpen(false)}
+        opened={modalOpen}
+        header={formatMessage(messages.selectFile)}
+      >
+        <Box mt="0px" p="24px">
+          {isDataRepositoryEnabled && (
+            <Box>
+              <Select
```

Comment:
Maybe we can hide or disable this select if `filteredFileOptions.length === 0`? Now, if there are no files, it just kind of does not work and it's not clear why

Reply:
Good catch! I'll add that. Thanks! :relaxed: